fire/SFNesterovMod.F90

```diff
@@ -0,0 +1,95 @@
+  implicit none
+  private
+
+  type, public, extends(fire_weather) :: nesterov_index 
+
+    contains 
+
+      procedure, public :: Init => init_nesterov_fire_weather
+      procedure, public :: Update => update_nesterov_index
+      procedure         :: calc_nesterov_index 
+
+  end type nesterov_index
+
+  real(r8), parameter :: min_precip_thresh = 3.0_r8 ! threshold for precipitation above which to 0.0 NI
+
+  contains 
+
+    subroutine init_nesterov_fire_weather(this)
+      !
+      !  DESCRIPTION:
+      !  Initializes class attributes
+      
+      ! ARGUMENTS
+      class(nesterov_index), intent(inout) :: this ! nesterov index extended class
+
+      ! initialize values to 0.0
+      this%fire_weather_index   = 0.0_r8
+
+    end subroutine init_nesterov_fire_weather
+
+    !-------------------------------------------------------------------------------------
+
+    subroutine update_nesterov_index(this, temp_C, precip, rh, wind)
+      !
+      !  DESCRIPTION:
+      !  Updates Nesterov Index
+      
+      ! ARGUMENTS
+      class(nesterov_index), intent(inout) :: this   ! nesterov index extended class
+      real(r8),              intent(in)    :: temp_C ! daily averaged temperature [degrees C]
+      real(r8),              intent(in)    :: precip ! daily precipitation [mm]
+      real(r8),              intent(in)    :: rh     ! daily relative humidity [%]
+      real(r8),              intent(in)    :: wind   ! daily wind speed [m/min]
+
+      if (precip > min_precip_thresh) then ! rezero NI if it rains
+        this%fire_weather_index = 0.0_r8
+      else 
+        ! Accumulate Nesterov index over fire season. 
+        this%fire_weather_index = this%fire_weather_index + &
+          this%calc_nesterov_index(temp_C, precip, rh)
+      end if 
+
+    end subroutine update_nesterov_index
+
+    !-------------------------------------------------------------------------------------
+
+    real(r8) function calc_nesterov_index(this, temp_C, precip, rh)
+      !
+      !  DESCRIPTION:
+      !  Calculates current day's Nesterov Index for a given input values
+
+      use SFParamsMod, only : SF_val_fdi_a, SF_val_fdi_b
+      
+      ! ARGUMENTS:
+      class(nesterov_index), intent(in) :: this   ! nesterov index extended class
+      real(r8),              intent(in) :: temp_C ! daily averaged temperature [degrees C]
+      real(r8),              intent(in) :: precip ! daily precipitation [mm]
+      real(r8),              intent(in) :: rh     ! daily relative humidity [rh]
+
+      ! LOCALS:
+      real(r8) :: yipsolon ! intermediate variable for dewpoint calculation
+      real(r8) :: dewpoint ! dewpoint
+
+      if (precip > min_precip_thresh) then ! NI is 0.0 if it rains
+        calc_nesterov_index = 0.0_r8
+      else 
+        ! Calculate dewpoint temperature 
+        yipsolon = (SF_val_fdi_a*temp_C)/(SF_val_fdi_b + temp_C) + log(max(1.0_r8, rh)/100.0_r8) 
+        dewpoint = (SF_val_fdi_b*yipsolon)/(SF_val_fdi_a - yipsolon) 
```

Comment:
This point is out of scope for this PR, but I just wanted to note that the parameters relevant to this approximation for dew point (which are [here](https://github.com/NGEET/fates/blob/main/parameter_files/fates_params_default.cdl#L1669) and [here](https://github.com/NGEET/fates/blob/main/parameter_files/fates_params_default.cdl#L1673)) seem more as approximations of thermodynamic constants than as tunable parameters, and also are very close to but not exactly [what wikipedia at least lists them as ](https://en.wikipedia.org/wiki/Dew_point#Calculating_the_dew_point). Given that, maybe it makes more sense to pull them out of the parameter file and move to [the physical constants file](https://github.com/NGEET/fates/blob/main/main/FatesConstantsMod.F90)?

Reply:
Sure sounds good, happy to do this.

Reply:
I guess we could migrate the exact constant values that we have in the parameter file now, and it would still be bit for bit. I'm still not sure I understand the source of the exact values of the constants we have now, but since they only differ from what is in the Wikipedia (and the reference pointed to there, which is https://doi.org/10.1175/BAMS-86-2-225)  in the fourth or so digit, I think it should be fine to leave as-is. Or we can change the numbers slightly so that they have a clear provenance, but then it won't be bit for bit. Or first the bit-for-bit change and then the non-bit-for-bit change.

Reply:
Okay yeah, I'm not sure what the provenance of them is either. The Thonicke paper approximates dewpoint temperature as Tmin-4, so we must have done this ourselves at some point.

Reply:
Okay I pulled those values out of the parameter file (but left them at what they were - we can change later?)

I also refactored this subroutine a bit to specifically calculate dewpoint as it's own function.